### PR TITLE
Complete Chitter Challenge

### PR DESCRIPTION
**Pull request by tbscanlon**
How I feel about my work on this challenge:
http://i.imgur.com/2qNqF2Y.png

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,22 +1,114 @@
...
+[Try it out here](https://chitter-june2017.herokuapp.com/).
 
-As usual please start by forking this repo.
+Some great features:
+- Almost entirely responsive using the magic of Bootstrap (ignore the non-collapsible navigation bar, I'll fix it soon&trade;)
+- Sign up for a user account!
+- Logging in and out works as well.
 
-We are going to write a little Twitter clone that will allow the users to post messages to a public stream.
+All tests are passing with 100% coverage. I ran into some issues unit testing my models, because I'm not sure how you're meant to stub out DataMapper functionality. I was able to test *most* of the behaviour of my models after referring to tests in one of the skills workshops we did this week, but it's still not as comprehensive as I would've liked. My comprehensive feature tests make up for this, however. I made a point of accounting for every edge case I could think of.
```
Comment (maryrosecook) [inline README.md]: Love these reflections!

File: app/app.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  get '/peeps/new' do
+    erb :'peeps/new'
+  end
+
+  post '/peeps/new' do
+    if current_user
+      Peep.create(message: params[:message], user_id: session[:user_id])
+      redirect to '/peeps'
+    else
+      flash.now[:errors] = "You must sign in or register to post a peep"
```
Comment (maryrosecook) [inline app/app.rb]: Can this line be made more declarative of what it does? At the moment, it seems to reveal its implementation, rather than its intention.

File: app/app.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  get '/users/new' do
+    erb :'/users/new'
+  end
+
+  post '/users/new' do
+    @user = User.new(username: params[:username], email: params[:email],
+                    password: params[:password],
+                    password_confirmation: params[:password_confirmation])
+    if @user.save
+      session[:user_id] = @user.id
```
Comment (maryrosecook) [inline app/app.rb]: Can this line be made more declarative of what it does? At the moment, it seems to reveal its implementation, rather than its intention.

File: app/models/peep.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+class Peep
+  include DataMapper::Resource
+
+  belongs_to :user, required: true
+
+  property :id, Serial
+  property :message, Text
+  property :created_at, DateTime
+
+  def time
```
Comment (maryrosecook) [inline app/models/peep.rb]: Does this code belong in the model?

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  property :id, Serial
+  property :username, String, required: true, unique: true
+  property :email, String, required: true, unique: true, format: :email_address
+  property :password_digest, Text
+
+  validates_confirmation_of :password
+
+  def self.authenticate(username, password)
+    user = first(username: username)
+    return user if user && BCrypt::Password.new(user.password_digest) == password
```
Comment (maryrosecook) [inline app/models/user.rb]: Do you need `return` here?

File: spec/unit/peep_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+require './app/models/user'
+
+describe Peep do
```
Comment (maryrosecook) [inline spec/unit/peep_spec.rb]: Love that you did unit tests!

File: app/app.rb [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    @peeps = Peep.all
+    erb :'peeps/peeps'
+  end
+
+  get '/peeps/new' do
+    erb :'peeps/new'
+  end
+
+  post '/peeps/new' do
+    if current_user
```
Comment (maryrosecook) [inline app/app.rb]: Is it possible to write this conditional in the form of a guard clause?
